src/ddmd/dstruct.d

```diff
@@ -323,10 +323,8 @@ extern (C++) class StructDeclaration : AggregateDeclaration
             userAttribDecl = sc.userAttribDecl;
         }
         else if (symtab && !scx)
-        {
-            semanticRun = PASSsemanticdone;
```

Comment:
Looks good, so the problem was that we set this to done here, even though we might be in a recursive call to StructDeclaration.semantic, with the outer semantic still ongoing on it's members, hence providing wrong semanticRun state info for any further members.